llvm/test/Transforms/InstCombine/fpclass-check-idioms.ll

```diff
@@ -75,6 +75,32 @@ define i1 @f32_fcnan(float %a) {
   ret i1 %res
 }
 
+define i1 @f32_fcnan_fcinf_strictfp(float %a) strictfp {
+; CHECK-LABEL: define i1 @f32_fcnan_fcinf_strictfp(
+; CHECK-SAME: float [[A:%.*]]) #[[ATTR0:[0-9]+]] {
+; CHECK-NEXT:    [[TMP1:%.*]] = call float @llvm.fabs.f32(float [[A]])
+; CHECK-NEXT:    [[CMP:%.*]] = fcmp ueq float [[TMP1]], 0x7FF0000000000000
```

Comment:
this shows a bug, we're not emitting a strict compare. doesn't need to be fixed in this patch though, existing issue

Reply:
I think this is an old bug but this patch has caused us to start hitting it in our floating-point library, I've posted a fix at #81498.